controller/tenant.go

```diff
@@ -175,22 +186,36 @@ func (c *TenantController) Update(ctx *app.UpdateTenantContext) error {
 	return ctx.Accepted()
 }
 
-func (c *TenantController) loadUserTenantConfiguration(token *jwt.Token, config openshift.Config) (openshift.Config, error) {
-	authHeader := token.Raw
-	if len(c.usersURL) > 0 {
-		url := strings.TrimSuffix(c.usersURL, "/") + "/user"
-		status, body, err := openshift.GetJSON(config, url, authHeader)
+// loadUserTenantConfiguration loads the tenant configuration for `auth`,
+// allowing for config overrides based on the content of his profile (in auth) if the user is allowed
+func (c *TenantController) loadUserTenantConfiguration(ctx context.Context, config openshift.Config) (openshift.Config, error) {
+	token := goajwt.ContextJWT(ctx)
+	// use the toggle service to verify that the user is allowed to change is profile based on the settings passed in the URL
+	if len(c.authURL) > 0 && c.toggleClient.IsEnabled(token, toggles.UserUpdateTenantFeature, false) {
+		log.Info(ctx, map[string]interface{}{"auth_url": c.authURL, "http_client_transport": reflect.TypeOf(c.httpClient.Transport)}, "retrieving user's profile...")
+		authClient, err := newAuthClient(ctx, c.httpClient, c.authURL)
 		if err != nil {
+			log.Error(ctx, map[string]interface{}{"auth_url": c.authURL}, "unable to parse auth URL")
 			return config, err
 		}
-		if status < 200 || status > 300 {
-			return config, fmt.Errorf("Failed to GET url %s due to status code %d", url, status)
+		resp, err := authClient.ShowUser(ctx, auth.ShowUserPath(), nil, nil)
+		if err != nil {
+			log.Error(ctx, map[string]interface{}{"auth_url": auth.ShowUserPath()}, "unable to get user info")
+			return config, errs.Wrapf(err, "failed to GET url %s due to error", auth.ShowUserPath())
+		}
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(resp.Body)
```

Comment:
Assuming you will remove lines 202-205 you may get a panic here because resp can be nil here.

Reply:
See my comment above: I should keep both blocks.